### ISSUE TITLE
Batch normalization as a composite op

### ISSUE THREAD

**Issue by jdh8** [tenstorrent/tt-metal#15449]
**Is your feature request related to a problem? Please describe.**
https://github.com/tenstorrent/pytorch2.0_ttnn/pull/415#discussion_r1857678373 shows that batch normalization should be a C++ composite op.  When we implement batch normalization in the compiler, the compiler converts batch normalization to a series of unary elementwise ops and `ttnn.reshape` that unsqueezes vectors to tensors to match dimensions.

**Describe the solution you'd like**
Implement batch normalization as a multiary (ternary?) composite op.
https://github.com/tenstorrent/tt-metal/tree/main/ttnn/cpp/ttnn/operations/eltwise

**Describe alternatives you've considered**
1. pytorch2.0_ttnn#415 already provides a suboptimal alternative, i.e. an implementation in the compiler.
2. I am not sure about the arity of batch normalization.

**Additional context**
If we want to implement training batch normalization as well, please take a look at [PyTorch decomposition algorithm](https://github.com/pytorch/pytorch/blob/ffb979032dc149b4c895526fe5b92d713ed7b1e1/torch/_decomp/decompositions.py#L1767-L1846).

**Comment by eyonland** [comment]
We would prefer to have this as a device op.  The plan is to complete #12253 by the end of this week.   Please let me know if you have any concerns.

**Comment by VirdhatchaniKN** [comment]
Training mode support is provided https://github.com/tenstorrent/tt-metal/pull/16432 and will be merged in a day.
Inference mode support is in progress

**Comment by VirdhatchaniKN** [comment]
> Training mode support is provided [#16432](https://github.com/tenstorrent/tt-metal/pull/16432) and will be merged in a day. Inference mode support is in progress

Inference mode support merged https://github.com/tenstorrent/tt-metal/pull/16592. Closing issue
